examples/groupchat/main.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+""" A group chat where user can talk any time implemented by agentscope. """
+from groupchat_utils import (
+    select_next_one,
+    filter_names,
+)
+
+import agentscope
+from agentscope.agents import UserAgent
+from agentscope.message import Msg
+from agentscope.msghub import msghub
+from agentscope.utils.common import timer
+
+USER_TIME_TO_SPEAK = 10
+
+
+def main() -> None:
+    """group chat"""
+    npc_agents = agentscope.init(
+        model_configs="./configs/model_configs.json",
+        agent_configs="./configs/agent_configs.json",
+    )
+
+    user = UserAgent()
+
+    agents = list(npc_agents) + [user]
+
+    hint = Msg(
+        name="Host",
+        content="This is a chat room and you can speak freely.",
+    )
+
+    rnd = 0
+    speak_list = []
+    with msghub(agents, announcement=hint):
+        while True:
+            try:
+                with timer(USER_TIME_TO_SPEAK):
+                    x = user()
+                    if x.content == "exit":
```

Comment:
If the input is very long and is not completed within the time limit,  it will need to be re-entered after the time limit is reached.

Maybe use a sub-thread to get the user's input and use a temp file to save the input until the user enters EOF.